### PR TITLE
[DONT REVIEW] JAMES-2865 MockSMTPServer rejecting/accepting + filtering messages

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

File: server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/MockSMTPServerTest.java [hunk 1]
```diff
@@ -386,10 +388,40 @@ void serverShouldNotReceiveMessageWhenSetRejected() throws Exception {
...
+                DATA,
+                Condition.MATCH_ALL,
+                Response.serverAccept(REQUESTED_MAIL_ACTION_NOT_TAKEN_450,
+                    "server cannot delivery message at after data step, will retry later"),
+                MockSMTPBehavior.NumberOfAnswersPolicy.anytime()));
+
+            sendMessageIgnoreError(mail1);
+
+            assertThat(mockServer.listReceivedMails())
+                .hasSize(1);
```
Comment (chibenwa) [inline server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/MockSMTPServerTest.java]: Assert the mail enveloppe

File: server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/MockSMTPServerTest.java [hunk 2]
```diff
@@ -23,6 +23,7 @@
 import static org.apache.james.mock.smtp.server.Fixture.BOB;
```
Comment (chibenwa) [inline server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/MockSMTPServerTest.java]: I'm missing a test with a "condition":

When "bob" 421 without reject, and alice 250, then a mail is stored with only alice in the rcpt field.

File: server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/Behaviors.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.james.mock.smtp.server.model.MockSMTPBehavior;
+import org.apache.james.mock.smtp.server.model.Response;
+import org.subethamail.smtp.RejectException;
+import org.subethamail.smtp.server.Session;
+
+import com.google.common.annotations.VisibleForTesting;
+
+interface Behaviors {
```
Comment (chibenwa) [inline server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/Behaviors.java]: Why do we need state tracking?

**Comment by trantienduchn** [comment]
misunderstand the requirement, need to rework

**Pull request closed without merging**